### PR TITLE
feat: update the samples of hyperparameter tuning in the public doc

### PR DESCRIPTION
**Pull request by gericdong**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #240582178 🦕

### CODE REVIEW

File: samples/model-builder/create_hyperparameter_tuning_job_sample.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    custom_job = aiplatform.CustomJob(
+        display_name='custom_job',
+        worker_pool_specs=worker_pool_specs,
+    )
+
+    hpt_job = aiplatform.HyperparameterTuningJob(
+        display_name=display_name,
+        custom_job=custom_job,
+        metric_spec={
```
Comment (ivanmkc) [inline samples/model-builder/create_hyperparameter_tuning_job_sample.py:54]: We usually pass in some of these as parameters to the function but I'm not sure how to include things like parameter specs without cluttering the parameter space.

Comment (ivanmkc) [reply samples/model-builder/create_hyperparameter_tuning_job_sample.py:54]: @dizcology do you have guidance or is this okay?

File: samples/model-builder/create_hyperparameter_tuning_job_sample.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    worker_pool_specs = [
+        {
+            "machine_spec": {
+                "machine_type": "n1-standard-4",
+                "accelerator_type": "NVIDIA_TESLA_K80",
+                "accelerator_count": 1,
+            },
+            "replica_count": 1,
+            "container_spec": {
+                "image_uri": container_uri,
```
Comment (ivanmkc) [inline samples/model-builder/create_hyperparameter_tuning_job_sample.py:39]: Seems weird to pass in `container_uri` when it's not used in the tuning job. 

@dizcology thoughts?

**Comment by ivanmkc** [comment]
In the future, please have one PR per sample.

File: samples/model-builder/create_hyperparameter_tuning_job_sample.py (hunk 2 above)
Comment (halio-g) [reply samples/model-builder/create_hyperparameter_tuning_job_sample.py:39]: Hptuning job creates sub jobs(trials) using the container image uri, it's actually used by the vertex hptuning jobs.

**Review by halio-g — APPROVED** [review]

**Review by ivanmkc — APPROVED** [review]
Let's get this in for now.

We can modify the parameters later if required.

**Pull request merged**